plugins/azure/fix_plugin_azure/resource/compute.py

```diff
@@ -557,6 +561,461 @@ class AzureDiskSecurityProfile:
+        query_parameters=[],
+        access_path=None,
+        expect_array=False,
+    )
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": K(None),
+        "offers": S("offers") >> Bend(AzurePricingOffers.mapping),
+        "graduated_offers": S("graduatedOffers") >> Bend(AzurePricingGraduatedOffers.mapping),
+    }
+    offers: Optional[AzurePricingOffers] = None
+    graduated_offers: Optional[AzurePricingGraduatedOffers] = None
+
+
+@define(eq=False, slots=False)
+class AzureDiskType(AzureResource, BaseVolumeType):
+    kind: ClassVar[str] = "azure_disk_type"
+    # Define api spec to collect as regional resources
+    api_spec: ClassVar[AzureApiSpec] = AzureApiSpec(
+        service="compute",
+        version="2023-01-01-preview",
+        path="",
+        path_parameters=["subscriptionId", "location"],
+        query_parameters=[],
+        access_path="Items",
+        expect_array=True,
+    )
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("skuName"),
+        "name": S("skuName"),
+        "full_name": S("armSkuName"),
+        "product_name": S("productName"),
+        "tier": S("skuName") >> F(lambda sku: sku.split(" ")[0]),
+        "redundancy": S("skuName") >> F(lambda sku: sku.split(" ")[1]),
+        "information_name": S("productName") >> F(lambda name: "_".join(name.split(" ")[:2])),
+        "ondemand_cost": S("unitPrice") >> F(lambda price: (price / 30) / 24),
+        "volume_type": (
+            (
+                (S("productName") >> F(lambda name: "_".join(name.split(" ")[:2])))
+                + K("_")
+                + (S("skuName") >> F(lambda sku: sku.split(" ")[1]))
+            )
+            >> MapValue(resource_group_map)
+        ).or_else(S("type")),
+        "volume_size": S("size"),
+        "volume_iops": S("maxIOPS"),
+        "volume_throughput": S("maxThroughput"),
+        "location": S("armRegionName"),
+    }
+    full_name: Optional[str] = None
+    product_name: Optional[str] = None
+    tier: Optional[str] = field(default=None, metadata={'description': 'Performance tier of the disk (e. G, p4, s10) as described here: https://azure. Microsoft. Com/en-us/pricing/details/managed-disks/. Does not apply to ultra disks.'})  # fmt: skip
+    redundancy: Optional[str] = None
+    information_name: Optional[str] = None
+    volume_iops: Optional[int] = None
+    volume_throughput: Optional[int] = None
+    volume_size: Optional[int] = None
+    location: Optional[str] = None
+    iops_price: Optional[float] = None
+    size_price: Optional[float] = None
+    throughput_price: Optional[float] = None
+    _is_provider_link: bool = False
+
+    def after_collect(self, builder: GraphBuilder, source: Json) -> None:
+        location = self.location
+        volume_type = self.volume_type
+
+        if location and volume_type in ("UltraSSD_LRS", "PremiumV2_LRS"):
+            for node in builder.graph.nodes:
+                if isinstance(node, AzureDiskTypePricing):
```

Comment:
assign them before the for loop.
pricing_nodes = builder.nodes(AzureDistTypePricing)